src/ansiblelint/yaml_utils.py

```diff
@@ -695,12 +695,19 @@ def write_comment(
                     ruamel.yaml.events.CollectionEndEvent,
                     ruamel.yaml.events.DocumentEndEvent,
                     ruamel.yaml.events.StreamEndEvent,
+                    ruamel.yaml.events.MappingStartEvent,
                 ),
             )
         ):
             # drop pure whitespace pre comments
             # does not apply to End events since they consume one of the newlines.
             value = ""
+        elif (
+            pre
+            and not value.strip()
+            and isinstance(self.event, ruamel.yaml.events.MappingStartEvent)
+        ):
+            value = self._re_repeat_blank_lines.sub("", value)
```

Comment:
Why would this work?

This is looking for a pure-whitespace comment before a map, and if there are `/n` or `/n/n`, it preserves them, but `/n/n/n` or longer gets replaced with `""`?

How does this behave when there are multiple blank lines between the tasks? Does it shorten it to only one?

Reply:
@cognifloyd I added this condition to remove the unnecessary spaces in pre comments, to make a scenario like this work: https://github.com/ansible/ansible-lint/blob/main/test/fixtures/formatting-before/fmt-1.yml#L2-L4. Since this is no longer done by the first check after adding MappingStartEvent.

Reply:
Ok. I finally understand how this is working. Thank you for the fix!

(Yes, I know this is already merged)